### DISCUSSION TITLE
Allow formatting from stdin

### DISCUSSION THREAD

**Discussion by silversquirl** [rome/tools#2855] [Suggestions]
Most other code formatters support reading code from stdin and outputting formatted code to stdout. This is a very useful standard interface for code formatters, and allows very easy integration with editors, such as through Kakoune's `formatcmd` option.

Since Rome can format multiple types of files, it needs to detect the filetype somehow; currently it uses the file extension for this, which would be incompatible with formatting from stdin. Options for handling this include a flag to specify the filetype from the command line (eg. `rome format --stdin --filetype js`), which would allow rome to benefit from the editor's detection; or copy what Prettier does and use a flag to specify the filename of the data provided on stdin (eg. `rome format --stdin --stdin-filename hello.js`).

**Comment by ematipico** [comment]
Great suggestion! Yes, I suppose it makes sense to have some argument like this!
